api/src/transactions.rs

```diff
@@ -1230,7 +1231,22 @@ impl TransactionsApi {
                 let mut user_transactions = Vec::new();
                 for transaction in transactions.into_iter() {
                     match transaction {
-                        Transaction::UserTransaction(user_txn) => user_transactions.push(*user_txn),
+                        Transaction::UserTransaction(user_txn) => {
+                            let mut txn = *user_txn;
+                            match &vm_status {
+                                VMStatus::Error {
+                                    message: Some(msg), ..
+                                }
+                                | VMStatus::ExecutionFailure {
+                                    message: Some(msg), ..
+                                } => {
+                                    txn.info.vm_status +=
```

Comment:
Isn't this populated by explain?

Reply:
This is exactly what we were trying to solve: execution status doesn’t have the message field so we have to get this info from vm status directly